services/controller/src/plz/controller/main.py

```diff
@@ -224,6 +229,14 @@ def get_output_files_entrypoint(execution_id):
     return Response(response, mimetype='application/octet-stream')
 
 
+def get_metadata(execution_id):
+    with results_storage.get(execution_id) as results:
+        if results is None:
+            raise ExecutionIdNotFound(
+                f'Metadata not found for execution id: {execution_id}')
```

Comment:
"ID", not "id".

Reply:
Probably more useful to be able to `raise ExecutionIdNotFound(execution_id)` and have that format the string accordingly.

Reply:
As to be able to recover the execution_id from the exception? I think I now did what you were looking for.